feat(gossipsub): make gossipsub modules private

tcoratger opened the pull request:
## Description

<!--
Please write a summary of your changes and why you made them.
This section will appear as the commit message after merging.
Please craft it accordingly.
For a quick primer on good commit mesages, check out this blog post: https://cbea.ms/git-commit/

Please include any relevant issues in here, for example:

Related https://github.com/libp2p/rust-libp2p/issues/ABCD.
Fixes https://github.com/libp2p/rust-libp2p/issues/XYZ.
-->

Resolves #3494.

## Notes & open questions

<!--
Any notes, remarks or open questions you have to make about the PR which don't need to go into the final commit message.
-->

## Change checklist

<!-- Please add a Changelog entry in the appropriate crates and bump the crate versions if needed. See <https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#development-between-releases>-->

- [x] I have performed a self-review of my own code
- [x] A changelog entry has been made in the appropriate crates

tcoratger committed:
make gossipsub modules private

tcoratger committed:
changelog PR ref

thomaseizinger reviewed (commented):
From memory, I think at least some types of the `metrics` module appear in the public API of `gossipsub::Behaviour`.

Can you spot check that and re-export them from the root crate?

tcoratger committed:
modify import modules to private

thomaseizinger committed:
Expose MetricsConfig

thomaseizinger committed:
Re-export subscription filters from root

thomaseizinger committed:
Merge branch 'master' into gossipsub_private

thomaseizinger committed:
Bump to next version and fix changelog

tcoratger committed:
Merge branch 'master' into gossipsub_private

tcoratger committed:
import Regex type in subscription_filter_priv tests

tcoratger committed:
Merge branch 'master' into gossipsub_private

thomaseizinger committed:
Fix test

thomaseizinger committed:
Merge branch 'gossipsub_private' of github.com:tcoratger/rust-libp2p into gossipsub_private

thomaseizinger commented on protocols/gossipsub/src/subscription_filter_priv.rs:
```diff
@@ -200,62 +200,20 @@ where
 mod test {
     use super::*;
     use crate::types::SubscriptionAction::*;
+    use regex::Regex;
```
```suggestion
```

thomaseizinger committed:
Fix clippy

tcoratger committed:
Merge branch 'master' into gossipsub_private

dependabot[bot] committed:
deps: bump h2 from 0.3.15 to 0.3.17

Pull-Request: #3788.

drHuangMHT committed:
refactor(identity): follow naming conventions for conversion methods

This PR renames some method names that don't follow Rust naming conventions or behave differently from what the name suggests:
- Enforce "try" prefix on all methods that return `Result`.
- Enforce "encode" method name for methods that return encoded bytes.
- Enforce "to_bytes" method name for methods that return raw bytes.
- Enforce "decode" method name for methods that convert encoded key.
- Enforce "from_bytes" method name for methods that convert raw bytes.

Pull-Request: #3775.

vnermolaev committed:
fix(gossipsub): gracefully disable handler on stream errors

Previously, we closed the entire connection upon receiving too many upgrade errors. This is unnecessarily aggressive. For example, an upgrade error may be caused by the remote dropping a stream during the initial handshake which is completely isolated from other protocols running on the same connection.

Instead of closing the connection, set `KeepAlive::No`.

Related: #3591.
Resolves: #3690.

Pull-Request: #3625.

thomaseizinger committed:
Merge branch 'master' into gossipsub_private

thomaseizinger reviewed (dismissed):
Thanks!

tcoratger committed:
Merge branch 'master' into gossipsub_private
